src/services/githubauth.service.js

```diff
@@ -124,6 +125,44 @@ export default class GithubAuthenticationService {
     return tokenCreated;
   }
 
+  /**
+   * First exchanges code and state for a github token.
+   * Then adds a github token for the given user into the githubTokens table. If a token already
+   * exists for the user, it will automatically update the old one.
+   * @param userId the user id to add the token for
+   * @param code the code to exchange for token
+   * @param state the state to verify identity
+   * @param username the github username of the user
+   * @returns {Object} the token that was created
+   */
+  async getAndSetGithubTokenForUser(userId, code, state, username) {
+    try {
+      const secrets = config.get('oauth_secrets');
+      const res = await fetch('https://cors-anywhere.herokuapp.com/https://github.com/login/oauth/access_token', {
```

Comment:
is cors anywhere secure?

Reply:
Also, the request might be better located in the `github.service.js` file. This is where we handle any interactions with the github client.

Reply:
I have made an issue to look into and find a better solution to this cors problem since I didn't feel it was closely tied to this issue, nor do I feel I have the most knowledge about this topic.